Modules/_interpretersmodule.c

```diff
@@ -1012,39 +1014,39 @@ interp_get_current(PyObject *self, PyObject *Py_UNUSED(ignored))
     return get_summary(interp);
 }
 
-PyDoc_STRVAR(get_current_doc,
-"get_current() -> (ID, whence)\n\
-\n\
-Return the ID of current interpreter.");
 
+/*[clinic input]
+_interpreters.get_main
+
+Return the ID of  main interpreter.
+[clinic start generated code]*/
 
 static PyObject *
-interp_get_main(PyObject *self, PyObject *Py_UNUSED(ignored))
+_interpreters_get_main_impl(PyObject *module)
+/*[clinic end generated code: output=9647288aff735557 input=a05bdf890db4b223]*/
 {
     PyInterpreterState *interp = _PyInterpreterState_Main();
     assert(_PyInterpreterState_IsReady(interp));
     return get_summary(interp);
 }
 
-PyDoc_STRVAR(get_main_doc,
-"get_main() -> (ID, whence)\n\
-\n\
-Return the ID of main interpreter.");
 
+/*[clinic input]
+_interpreters.set___main___attrs
+    id: object
+    ns as updates: object(subclass_of='&PyDict_Type')
```

Comment:
The name is "updates".

Reply:
Currently only "updates" is accepted. But it is not documented, so we have some freedom. The fact that "updates" is not documented and "ns" does not work can be interpreted in favor of making them positional-only. @ericsnowcurrently?

Reply:
I've reverted this to use `updates` per the `PyArg_ParseTupleAndKeywords` call. `ns` might be a better fit, but we can defer that change to a follow-up PR.